indexing/src/main/scala/org/allenai/common/indexing/BuildCorpusIndexRunner.scala

```diff
@@ -1,27 +1,33 @@
 package org.allenai.common.indexing
 
-import com.typesafe.config.ConfigFactory
+import com.typesafe.config.{ Config, ConfigFactory }
+
+import java.io.File
 
 /** Indexing main object. Configuration specified in indexing.conf in org.allenai.common.indexing.
   * See common/Readme for details.
   */
 object BuildCorpusIndexRunner extends App {
 
-  case class IndexConfig(indexConfigName: String = null)
+  case class IndexConfig(indexConfigName: String = null, configOverrideFile: Option[File] = None)
 
   val parser = new scopt.OptionParser[IndexConfig]("BuildCorpusIndexRunner") {
     head("BuildCorpusIndexRunner")
     note("See common/Readme for details on how to use this.\n")
     help("help") text ("prints this usage text")
-    arg[String]("<indexConfigName>...") required () action { (x, c) =>
+    opt[String]("index-name") required () valueName ("<string>") action { (x, c) =>
       c.copy(indexConfigName = x)
     } text ("name of configuration to use")
+    opt[File]("config-overrides-file") valueName ("<file>") action { (x, c) =>
+      c.copy(configOverrideFile = Some(x))
+    } text ("Config file with overrides if any")
   }
 
   parser.parse(args, IndexConfig()).foreach(config => {
     val rootConfig = ConfigFactory.parseResources(getClass, "indexing.conf").resolve()
+    val configOverrides = config.configOverrideFile map { f => ConfigFactory.parseFile(f) }
```

Comment:
I would personally put the pattern matching here that way you don't have to change the signature for `BuildCorpusIndex`. I mean something like:
```
val originalConfig = rootConfig.getConfig(s"org.allenai.common.indexing.${config.indexConfigName}")
val finalConfig = configOverrides match {
    case Some(overrides) => overrides.withFallback(originalConfig)
    case None => originalConfig
}
new BuildCorpusIndex(finalConfig.resolve()).buildElasticSearchIndex()
```
But up to you!